src/views/AddItem.jsx

```diff
@@ -1,3 +1,81 @@
+import { useState } from 'react';
+import { addItem } from '../api';
+import { useStateWithStorage } from '../utils';
+
+import './AddItem.css';
+
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	const [itemToAdd, setItemToAdd] = useState({});
+	const [listToken, setListToken] = useStateWithStorage(
+		'my test list',
+		'tcl-shopping-list-token',
+	);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		try {
+			const docRef = await addItem(listToken, {
+				itemName: itemToAdd.itemName,
+				daysUntilNextPurchase: itemToAdd.buyingFrequency,
+			});
+
+			docRef && alert(`${itemToAdd.itemName} was saved to the database`);
+		} catch (err) {
+			console.log(err);
+			err && alert('item not saved, pls try again');
+		}
+	};
+
+	const handleChange = (e) => {
+		const { name, value } = e.target;
+		setItemToAdd({ ...itemToAdd, [name]: value });
+	};
```

Comment:
Great use of destructuring and object manipulation, love it  ❤️. For better practices and to avoid any issues use the previous state.
`setItemToAdd((prevState) => ({ ...prevState, [name]: value }));`